tests/jdom/newbutton.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { suite, test } from "mocha"
+import { ButtonEvent, ButtonReg } from "../../src/jdom/constants"
+import ButtonServer from "../../src/servers/buttonserver"
+import { ServiceTester } from "../../src/tstester/servicewrapper"
+import { FastForwardBusTester, FastForwardTestDriver } from "./newtester"
+
+suite("button server", () => {
+    // Tolerances are set to 50ms as a typical register update interval, plus another 50ms for event alignment
+    test("fires edge events after changing state", async function () {
+        await FastForwardBusTester.withTestBus(async bus => {
```

Comment:
not a fan of nested static function. let's move this as a top level exported function